### ISSUE TITLE
Add support for overriding styles with Emotion or StyledComponents

### ISSUE THREAD

**Issue by yadielar** [iamhosseindhv/notistack#461]
## Expected Behavior
There should be a way to easily override Snackbar styles with styled component libraries like Emotion with access to the app's theme. Something like:

```js
import { DefaultSnackbarContent } from 'notistack';

const StyledSnackbarContent = styled(DefaultSnackbarContent)(({ theme }) => ({
  '&.SnackbarItem-variantSuccess': {
    backgroundColor: '#fff',
    color: '#000',
    borderLeft: `solid 5px ${theme.palette.success.main}`,
  },
  '&.SnackbarItem-variantInfo': {
    backgroundColor: '#fff',
    color: '#000',
    borderLeft: `solid 5px ${theme.palette.primary.main}`,
  },
  '&.SnackbarItem-variantError': {
    backgroundColor: '#fff',
    color: '#000',
    borderLeft: `solid 5px ${theme.palette.error.main}`,
  },
  '&.SnackbarItem-variantWarning': {
    backgroundColor: '#fff',
    color: '#000',
    borderLeft: `solid 5px ${theme.palette.warning.main}`,
  },
}));

function App() {
  return (
    <SnackbarProvider  
      maxSnack={3}
      content={(...snackProps) => <StyledSnackbarContent {...snackProps} />}
    >
      <Routes />
    </SnackbarProvider>
  );
}
```

## Current Behavior
Currently, there's no way to override variants styles easily with Emotion, it only supports [Material UI classes with JSS](https://iamhosseindhv.com/notistack/demos#custom-variants).

## Context
Support for style overrides with Emotion is particularly needed when using MUI v5, since they've adopted Emotion as the base styling engine and have deprecated JSS. In their docs, they recommend [using the default `Mui*` class names inside styled components to style deeper elements.](https://mui.com/guides/interoperability/#deeper-elements-3)

For notistack v3, having this convenient API would be a good option with the same spirit as the current `classes` prop, for those who want to customize the default `SnackbarContent` only slightly without needing to create completely custom components for each variant.

**Comment by Artofaj** [comment]
I am currently working on a project that relies on this. If you need any help, please let me know, I would be more than happy to assist.

Amazing project btw.!

**Comment by iamhosseindhv** [comment]
So we're talking about a scenario where developers want to "slightly" customise Snackbars, rather than defining brand new custom components as described [in the docs](https://notistack.com/examples/advanced/custom-component).

In which case, would it be beneficial to export this component from notistack?, and use it as such:
https://github.com/iamhosseindhv/notistack/blob/alpha/src/ui/MaterialDesignContent/MaterialDesignContent.tsx


```tsx
import { MaterialDesignContent } from 'notistack'

const StyledContent = styled(MaterialDesignContent)`
   // customise
`


<SnackbarProvider
   Components={{
       default: StyledContent,
       // ...
   }}
>

```

Let me know what you think

**Comment by d4rky-pl** [comment]
@iamhosseindhv re-exporting the `MaterialDesignContent` so it can be wrapped in custom `styled` would be lovely 🙏

**Comment by giorgiabosello** [comment]
Any updates about this?

**Comment by koooge** [comment]
Hi there,
I am using `classes` prop with `makeStyles()` of @mui/styles for now but [@mui/styles was marked as deprecated](https://mui.com/system/styles/basics/) became deprecated. Snackbar are applied Theme-aware styles of Mui in my case. styled-component or `sx` props support would be great. I want the prop to accept some levels(success, error, warning, info) like `iconVariant`.

**Comment by giorgiabosello** [comment]
Any update on this?

**Comment by snoop088** [comment]
I think that simple modifications to colours for the **default**, **success**, **error**, **warning** and **info** snackbars would be really helpful if done in simple way. 
Probably large percent of modifications necessary to this component is being able to fit it in some kind of colour theme that you are using across your app. You already have a good way to override the complete component with a custom one, but in many situations in my opinion people would only want to change the SnackbarContent error background to some other 'red' that they are using for errors across their app. Why would you need to define a custom component just for changing the colours..

hopefully this makes sense :)

**Comment by iamhosseindhv** [comment]
A new version of notistack v3 has been released  (`v3.0.0-alpha.13` / `npm i notistack@alpha`) - so you can now do something like:

```tsx
import { MaterialDesignContent } from 'notistack'

const StyledMaterialDesignContent = styled(MaterialDesignContent)(() => ({
  '&.notistack-MuiContent-success': {
    backgroundColor: '#2D7738',
  },
  '&.notistack-MuiContent-error': {
    backgroundColor: '#970C0C',
  },
}));

<SnackbarProvider
  Components={{
    success: StyledMaterialDesignContent,
    error: StyledMaterialDesignContent,
  }}
>
```

[Customization page](https://notistack.com/features/customization#override-styles) on documentation website has also been updated, explaining how to "Override Styles":

> ... You can import notistack's Material Snackbar and override its styles. The class names are called: `notistack-MuiContent` and `notistack-MuiContent-<variant_name>`.
